viper.go

```diff
@@ -588,36 +589,58 @@ func (v *Viper) searchMap(source map[string]interface{}, path []string) interfac
 // in their keys).
 //
 // Note: This assumes that the path entries and map keys are lower cased.
-func (v *Viper) searchMapWithPathPrefixes(source map[string]interface{}, path []string) interface{} {
+func (v *Viper) searchMapWithPathPrefixes(source interface{}, path []string) interface{} {
 	if len(path) == 0 {
 		return source
 	}
 
 	// search for path prefixes, starting from the longest one
 	for i := len(path); i > 0; i-- {
 		prefixKey := strings.ToLower(strings.Join(path[0:i], v.keyDelim))
+		if sourceSlice, ok := source.([]interface{}); ok {
+			//if the prefixKey is a number which is not out of bounds of the slice
+			if index, err := strconv.Atoi(prefixKey); err == nil && len(sourceSlice) > index {
```

Comment:
By inverting this condition and returning early, we can avoid the arrow pattern.